server/sublist.go

```diff
@@ -1735,3 +1737,44 @@ func getAllNodes(l *level, results *SublistResult) {
 		getAllNodes(n.next, results)
 	}
 }
+
+// interestBasedIntersection will match all items in the given subject tree that
+// have interest expressed in the given sublist. The callback will only be called
+// once for each subject, regardless of overlapping subscriptions in the sublist.
+func interestBasedIntersection[T any](st *stree.SubjectTree[T], sl *Sublist, cb func(subj []byte, entry *T)) {
```

Comment:
Should this be a function on a *sublist? Or that not possible with generics?

If not make it public? IntersectStree()?

Reply:
Not possible with generics, but yep, happy to rename it to that.